Origin: upstream https://github.com/varnishcache/varnish-cache/commit/26d52615004730eb28ce603930600d44cafd8202
Description: Fix memory leak of vary string on stevedore alloc fail. This fix is included
in varnish 6.0.x and 6.4.x so it can be safely removed.
commit 26d52615004730eb28ce603930600d44cafd8202
Author: Martin Blix Grydeland <martin@varnish-software.com>
Date:   Wed Mar 14 13:52:36 2018 +0100

    Fix memory leak of vary string on stevedore alloc fail
    
    If the stevedore failed the object creation, we would leak the temporary
    VSB holding the computed vary string. This patch frees it.
    
    Problem exists in 4.1 and later.

Index: varnish4/bin/varnishd/cache/cache_fetch.c
===================================================================
--- varnish4.orig/bin/varnishd/cache/cache_fetch.c
+++ varnish4/bin/varnishd/cache/cache_fetch.c
@@ -135,8 +135,12 @@ vbf_beresp2obj(struct busyobj *bo)
 	if (bo->uncacheable)
 		bo->fetch_objcore->flags |= OC_F_PASS;
 
-	if (!vbf_allocobj(bo, l))
+	if (!vbf_allocobj(bo, l)) {
+		if (vary != NULL)
+			VSB_destroy(&vary);
+		AZ(vary);
 		return (-1);
+	}
 
 	if (vary != NULL) {
 		AN(ObjSetAttr(bo->wrk, bo->fetch_objcore, OA_VARY, varyl,
